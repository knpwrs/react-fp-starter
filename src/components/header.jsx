import React from 'react';
import g from 'glamorous';
import { pure } from 'recompose';
import StyledLink from './styled-link';

const HeaderLink = StyledLink({
  textDecoration: 'none',
  marginLeft: '1em',
  fontSize: '2em',
  color: 'grey',
  ':hover': {
    color: 'black',
  },
});

const Header = () => (
  <g.Header
    display="flex"
    alignItems="center"
  >
    <h1>React FP Starter</h1>
    <g.Nav
      marginLeft="auto"
    >
      <HeaderLink to="/home">Home</HeaderLink>
      <HeaderLink to="/about">About</HeaderLink>
    </g.Nav>
  </g.Header>
);

export default pure(Header);
