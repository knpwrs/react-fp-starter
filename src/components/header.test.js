/* eslint-disable react/jsx-filename-extension */
// https://github.com/avajs/ava/issues/631
import test from 'ava';
import R from 'ramda';
import React from 'react';
import render from 'react-test-renderer';
import { Link } from 'react-router-dom';
import { withContext } from 'recompose';
import { Bare as Header } from './header';

const routerContext = withContext(Link.contextTypes, () => ({
  router: {
    history: {
      push: R.T,
      replace: R.T,
      createHref: R.T,
    },
  },
}));

test('counter', (t) => {
  const H = routerContext(Header);
  const tree = render.create(<H />);
  t.snapshot(tree);
});
