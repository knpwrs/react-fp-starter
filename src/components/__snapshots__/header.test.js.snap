// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter 1`] = `
<header
  className="css-12alag6"
>
  <h1>
    React FP Starter
  </h1>
  <nav
    className="css-omocl"
  >
    <a
      className="css-j5mfvq"
      href={true}
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="css-j5mfvq"
      href={true}
      onClick={[Function]}
    >
      About
    </a>
  </nav>
</header>
`;
